webapp/channels/src/components/post/post_component.tsx

```diff
@@ -130,24 +130,43 @@ const PostComponent = (props: Props): JSX.Element => {
     const [fileDropdownOpened, setFileDropdownOpened] = useState(false);
     const [fadeOutHighlight, setFadeOutHighlight] = useState(false);
     const [alt, setAlt] = useState(false);
+    const [hasReceivedA11yFocus, setHasReceivedA11yFocus] = useState(false);
 
     const isSystemMessage = PostUtils.isSystemMessage(post);
     const fromAutoResponder = PostUtils.fromAutoResponder(post);
 
     useEffect(() => {
-        if (props.shouldHighlight) {
+        if (shouldHighlight) {
             const timer = setTimeout(() => setFadeOutHighlight(true), Constants.PERMALINK_FADEOUT);
             return () => {
                 clearTimeout(timer);
             };
         }
         return undefined;
-    }, [props.shouldHighlight]);
+    }, [shouldHighlight]);
 
     const handleA11yActivateEvent = () => setA11y(true);
     const handleA11yDeactivateEvent = () => setA11y(false);
     const handleAlt = (e: KeyboardEvent) => setAlt(e.altKey);
 
+    const handleA11yKeyboardFocus = useCallback((e: KeyboardEvent) => {
+        if (e.key === Constants.KeyCodes.TAB[0] && e.shiftKey && shouldHighlight && !hasReceivedA11yFocus) {
```

Comment:
Nit: But may be instead we can use https://github.com/mattermost/mattermost-server/blob/a85a7c84220ee9f9c83d729fa5dbfb24abd01b20/webapp/channels/src/utils/keyboard.ts#L16

Reply:
Didn't know this existed, thank you!